packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -228,6 +227,63 @@ export class Alarm extends Resource implements IAlarm {
   }
 }
 
+function renderAlarmMetric(metric: IMetric) {
+  return dispatchMetric(metric, {
+    withStat(st) {
+      return dropUndef({
+        dimensions: st.dimensions,
+        namespace: st.namespace,
+        metricName: st.metricName,
+        period: st.period?.toSeconds(),
+        statistic: renderIfSimpleStatistic(st.statistic),
+        extendedStatistic: renderIfExtendedStatistic(st.statistic),
+        unit: st.unitFilter,
+      });
+    },
+
+    withExpression() {
+      // Expand the math expression metric into a set
+      const mset = new MetricSet<boolean>();
+      mset.addTopLevel(true, metric);
+
+      let eid = 0;
+      function uniqueMetricId() {
+        return `expr_${++eid}`;
+      }
+
+      return {
+        metrics: mset.entries.map(entry => dispatchMetric(entry.metric, {
+          withStat(stat, conf) {
+            return {
+              metricStat: {
+                metric: {
+                  metricName: stat.metricName,
+                  namespace: stat.namespace,
+                  dimensions: stat.dimensions,
+                },
+                period: stat.period.toSeconds(),
+                stat: stat.statistic,
+                unit: stat.unitFilter,
+              },
+              id: entry.id || uniqueMetricId(),
+              label: conf.renderingProperties?.label,
+              returnData: entry.tag ? undefined : false, // Tag stores "primary" attribute, default is "true"
+            };
+          },
+          withExpression(expr, conf) {
+            return {
+              expression: expr.expression,
+              id: entry.id || uniqueMetricId(),
+              label: conf.renderingProperties?.label,
+              returnData: entry.tag ? undefined : false, // Tag stores "primary" attribute, default is "true"
+            };
+          },
+        }) as CfnAlarm.MetricDataQueryProperty)
```

Comment:
Are you sure? That's not entirely what happens. I use this to guide IntelliSense to tell me what properties I need to put on an otherwise untyped `{ ... }` object literal.

In fact if I change this to read something else, like `as CfnAlarm.MetricProperty` the type checker starts spewing errors at me, so it's definitely not a simple overriding cast.

I'm inclined to leave it.

Reply:
If you include a property that is *not* part of `CfnAlarm.MetricDataQueryProperty` (for example, because you typo'd an optional field's name), does it fail? Or does it silently accept it?